File: src/main/scala/com/abaddon83/cardsGames/burracoGames/domainModels/Scale.scala
package com.abaddon83.cardsGames.burracoGames.domainModels

import com.abaddon83.cardsGames.shares.decks.Suits.Suit
import com.abaddon83.cardsGames.shares.decks.{Card, Ranks}


trait Scale {
  protected val cards: List[Card]
  protected val suit: Suit


  def addCards(cardsToAdd: List[Card]): Scale

  def numCards():Int = {
    cards.size
  }

  protected def validateNewCards(cards: List[Card]): Scale


}

File: src/main/kotlin/com/abaddon83/burracoGame/commandModel/models/burracos/Scale.kt
package com.abaddon83.burracoGame.commandModel.models.burracos

import com.abaddon83.burracoGame.commandModel.models.decks.Card
import com.abaddon83.burracoGame.commandModel.models.decks.Suits

abstract class Scale: Burraco() {

    fun numCards():Int = cards.size

    abstract fun addCards(cardsToAdd: List<Card>): Scale
    protected abstract val suit: Suits.Suit
    protected abstract fun validateNewCards(cardsToValidate: List<Card>): List<Card>

}